var/spack/repos/builtin/packages/imagemagick/package.py

```diff
@@ -26,15 +28,20 @@ class Imagemagick(AutotoolsPackage):
     depends_on("freetype")
     depends_on("fontconfig")
     depends_on("libtiff")
-    depends_on("ghostscript")
-    depends_on("ghostscript-fonts")
+    depends_on("ghostscript", when="+ghostscript")
+    depends_on("ghostscript-fonts", when="+ghostscript")
     depends_on("libsm")
     depends_on("pkgconfig", type="build")
 
     def configure_args(self):
+        args = []
         spec = self.spec
-        gs_font_dir = join_path(spec["ghostscript-fonts"].prefix.share, "font")
-        return ["--with-gs-font-dir={0}".format(gs_font_dir)]
+        if spec.satisfies("+ghostscript"):
+            gs_font_dir = join_path(spec["ghostscript-fonts"].prefix.share, "font")
+            args.append("--with-gs-font-dir={0}".format(gs_font_dir))
```

Comment:
Curious.  Do you also need `--with-gslib` or is that "automatic" with `-with-gs-font-dir`?

Reply:
It defaults to `with`, but you're right that it is better to be explicit.
```suggestion
        if spec.satisfies("+ghostscript"):
            args.append("--with-gslib")
            gs_font_dir = join_path(spec["ghostscript-fonts"].prefix.share, "font")
            args.append("--with-gs-font-dir={0}".format(gs_font_dir))
```